src/Magento/FunctionalTestingFramework/Console/CleanProjectCommand.php

```diff
@@ -0,0 +1,109 @@
+<?php
+// @codingStandardsIgnoreFile
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types = 1);
+
+namespace Magento\FunctionalTestingFramework\Console;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+class CleanProjectCommand extends Command
+{
+    const CONFIGURATION_FILES = [
+        TESTS_BP . DIRECTORY_SEPARATOR . 'codeception.yml', // codeception.yml file for top level config
+        TESTS_BP . DIRECTORY_SEPARATOR . 'tests' . DIRECTORY_SEPARATOR . 'functional.suite.yml', // functional.suite.yml for test execution config
+        FW_BP . '/src/Magento/FunctionalTestingFramework/_generated', // Acceptance Tester Actions generated by codeception
+        FW_BP . '/src/Magento/FunctionalTestingFramework/AcceptanceTester.php' // AcceptanceTester Class generated by codeception
+    ];
+
+    const GENERATED_FILES = [
+        TESTS_MODULE_PATH . '/_generated'
+    ];
+
+    /**
+     * Configures the current command.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this->setName('reset')
+            ->setDescription('This command will clean any configuration files from the environment (not including .env), as well as any generated artifacts.')
+            ->addOption('hard', null, InputOption::VALUE_NONE, "parameter to force reset of configuration files.");
+    }
+
+    /**
+     * Executes the current command.
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return void
+     * @throws \Symfony\Component\Console\Exception\LogicException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $isHardReset = $input->getOption('hard');
+        $fileSystem = new Filesystem();
+        $finder = new Finder();
+        $finder->files()->name('*.php')->in(FW_BP . '/src/Magento/FunctionalTestingFramework/Group/');
+
+        $beforeState = $this->getFileSystemState($finder); // used only upon verbose output but useful for capturing debug state
+
+        // delete config files if user specifies a hard reset
+        if ($isHardReset) {
+            $fileSystem->remove(self::CONFIGURATION_FILES);
+        }
+
+        // delete the files mftf generates during test execution in TESTS_BP
+        $fileSystem->remove(self::GENERATED_FILES);
+
+        // delete any suite based group files
+        $fileSystem->remove($finder);
+
+        if ($output->isVerbose()) {
+            // build a list of the paths
+            $afterState = $this->getFileSystemState($finder);
+            $deletedFiles = array_diff($beforeState, $afterState);
```

Comment:
I am not sure I understand this comment. From what I understand if a file does not exist and filesystem attempts to remove there is no exception, how can I rely on this list? Additionally I am deleting php files in certain directories, that will not tell me explicit file deletion to output.

Reply:
Maybe I'm missing something, but this is how I understood it should work.
```
protected function execute(InputInterface $input, OutputInterface $output)
    {
        $fileSystem = new Filesystem();

        // delete the files mftf generates during test execution in TESTS_BP
        if (is_dir(self::GENERATED_FILES)) {
            $fileSystem->remove(self::GENERATED_FILES);
            if ($output->isVerbose()) {
                $output->writeln("Folder with generated files deleted.");
            }
        }
        
        $finder = new Finder();
        $files = $finder->files()->name('*.php')->in(FW_BP . '/src/Magento/FunctionalTestingFramework/Group/');
        foreach ($files as $file) {
            $path = $file->getRealPath();
            $fileSystem->remove($file);
            if ($output->isVerbose()) {
                $output->writeln(sprintf('File %s deleted.', $path));
            }
        }

        // delete config files if user specifies a hard reset
        $isHardReset = $input->getOption('hard');
        if ($isHardReset) {
            foreach (self::CONFIGURATION_FILES as $configFile) {
                if (file_exists($configFile)) {
                    $fileSystem->remove($configFile);
                    if ($output->isVerbose()) {
                        $output->writeln("Config file $configFile deleted.");
                    }
                }
            }
        }

        $output->writeln('<info>Project cleaned successfully.<info>');
    }
```

Reply:
Okay I see what you are saying, I didn't want to explicitly loop the files since the filesystem command will go ahead and delete the whole group on its own but you're right if we want to have the explicit print we'll have to loop the files ourselves and output.